File: code-dist-2019-10/Python/rei2p7/least_square_demo.py
import cvxpy as cp
import numpy as np
r = 20
np.random.seed(1)
A = np.hstack((np.random.randn(r,1), np.ones([r,1])))
c = A[:,0]
b = (10.0*np.random.randn() * c) + \
        + (0.5*np.random.randn(r))
x    = cp.Variable(2)
obj  = cp.Minimize( sum(cp.square(A*x-b)) )
P = cp.Problem(obj)
P.solve(verbose=True)
print(x.value)

File: code-dist-2019-10/MATLAB/rei2p7/least_square_demo.m
r = 20;
rng(0,'twister');
A = [randn(r,1), ones(r,1)];
b = ((10*randn)*A(:,1)) + (0.5*randn(r,1));
cvx_begin
    variable x(2,1)
    minimize( (A*x - b)' * (A*x - b) )
cvx_end
